bots/sg_single_rule_delete.py

```diff
@@ -202,10 +203,10 @@ def run_action(boto_session, rule, entity, params):
                     text_output = text_output + 'Security Group ingress rule from port %s to port %s successfully added\n' % (
                         upper_port_number, upper_port_number_to)
 
-    if split == True:
-        if protocol_to_remove:
-            lower_port_number = -1
-            upper_port_number_to = 0
+    if split:
+        if '-1' == protocol_to_remove:  # -1 specify all protocols - bug fixed : '-1' cannot use as boolean
+            lower_port_number = 0
+            upper_port_number_to = -1
```

Comment:
I don't know if it doesn't break other things

Reply:
I tried to debug it as much as I could, but I can't be sure what the writer of the Bot meant exactly in this 'if ' condition. I believe this is what he meant
protocol_to_remove defined only down here, I think the condition is to check if protocol_to_remove defined - so enter the scope. And it's the same to ask if its equals to '-1' because its the only time it defined

# In case we want to delete the SG with protocol = 'All' traffic
if (protocol == 'ALL'):
if (split == True):
rule['port'] = 0
rule['portTo'] = 65535
protocol_to_remove = '-1' # -1 specify protocol = All
protocol = 'tcp'
else:
port = -1
portTo = -1
protocol = '-1'
split = False
rule_to_delete = rule
break`